types/lodash/index.d.ts

```diff
@@ -12465,71 +12465,69 @@ declare namespace _ {
          * @param defaultValue The value returned if the resolved value is undefined.
          * @return Returns the resolved value.
          */
-        get<TResult>(
-            object: any,
-            path: Many<StringRepresentable>,
-            defaultValue: TResult
-        ): TResult;
+        // when object is empty - definitely default value will be returned
+        get<TDefault>(
+            object: null | undefined,
+            path: Many<StringRepresentable>, // ** Many<StringRepresentable> instead of any
+            defaultValue: TDefault // ** no longer optional
+        ): TDefault;
+
+        // ** same as above, but with no default value
+        get(
+            object: null | undefined,
+            path: Many<StringRepresentable>
+        ): undefined;
 
-        /**
-         * @see _.get
-         */
-        get<TObject extends object, TKey extends keyof TObject>(
-            object: TObject | null | undefined,
+        // when depth of object is 1 and path to value is right - or value, or default will be returned (in case value is undefined)
+        get<TObject extends object, TKey extends keyof TObject, TDefault>(
+            object: TObject,
             path: TKey | [TKey],
-            defaultValue: TObject[TKey]
-        ): TObject[TKey];
+            defaultValue: TDefault // ** no longer optional
+        ): TObject[TKey] | TDefault;
```

Comment:
Originally I included this overload, but then I decided against it because it's wrong if `TObject[TKey]` is possibly undefined - then `undefined` could be included in the return type, even though it would never be the return value.

Example:

```
let obj: { a: string | undefined };
let result = _.get(obj, "a", "default"); // Return type is string | undefined, but should be string
```

I'm a little on the fence on whether this overload should be included. Maybe it's better to return *almost* the right type (but include `undefined` when we shouldn't) than it is to return `any`. But on the other hand, people will probably think this is a bug and request us to fix it.